Address remaining elements of Jon's review (#10)
ietf-scitt/draft-birkholz-cose-cometre-ccf-profile: draft-birkholz-cose-receipts-ccf-profile.md

normative:
  RFC9162: certificate-transparency-v2

>   I-D.draft-ietf-cose-merkle-tree-proofs: cose-receipts

Author: I think this is right, but I am not sure how to check that references work properly. I am willing to learn though.

Reviewer: There are many ways, including https://author-tools.ietf.org/
The path I tend to use, which I find more iterative and local is to use the `make` tooling.
You'll need a current node environment. I tend to use nvm:

```
nvm use 18
make
```
Assuming make doesn't generate any errors, browse the generated HTML locally: 
 file://wsl.localhost/Ubuntu/home/steve/go/src/github/achamayou/draft-birkholz-cose-cometre-ccf-profile/draft-birkholz-cose-receipts-ccf-profile.html 

Each file change, locally, you can then just re-run make, and refresh your browser... voila

There are a few nit warnings for longer than 72 bytes for samples
```
(160): Warning: Too long line found (L185), 7 characters longer than 72 characters:
     internal-transaction-hash: bstr .size 32 ; a string of HASH_SIZE(32) bytes
/home/steve/go/src/github/achamayou/draft-birkholz-cose-cometre-ccf-profile/draft-birkholz-cose-receipts-ccf-profile.xml(160): Warning: Too long line found (L186), 6 characters longer than 72 characters:
     internal-evidence: tstr .size (1..1024)  ; a string of at most 1024 bytes
/home/steve/go/src/github/achamayou/draft-birkholz-cose-cometre-ccf-profile/draft-birkholz-cose-receipts-ccf-profile.xml(160): Warning: Too long line found (L187), 7 characters longer than 72 characters:
     data-hash: bstr .size 32                 ; a string of HASH_SIZE(32) bytes
/home/steve/go/src/github/achamayou/draft-birkholz-cose-cometre-ccf-profile/draft-birkholz-cose-receipts-ccf-profile.xml(175): Warning: Too long line found (L210), 11 characters longer than 72 characters:
     hash: bstr .size 32; hash of the proof element (string of HASH_SIZE(32) bytes)
/home/steve/go/src/github/achamayou/draft-birkholz-cose-cometre-ccf-profile/draft-birkholz-cose-receipts-ccf-profile.xml(210): Warning: Too long line found (L270), 10 characters longer than 72 characters:
  let proof = inclusion_receipt.unprotected_headers[INCLUSION_PROOF_LABEL] or fail
```

...but the reference change is correct.

Author: @SteveLasker thank you for the pointers, I have found that it is also necessary to `gem install kramdown-rfc` to get the Makefile working. I will try to fix the warnings.

Author: ```
~/draft-birkholz-cose-cometre-ccf-profile$ make
lib/deps.mk:192: ruby bundler not installed; skipping bundle install
draft-birkholz-cose-receipts-ccf-profile: kramdowm-rfc ... OK
draft-birkholz-cose-receipts-ccf-profile: xml2rfc-txt ... OK
draft-birkholz-cose-receipts-ccf-profile: xml2rfc-html ... OK
rm draft-birkholz-cose-receipts-ccf-profile.xml
```

all fixed now

>   I-D.draft-ietf-cose-merkle-tree-proofs: cose-receipts

Reviewer 2: ```suggestion
  I-D.ietf-cose-merkle-tree-proofs: cose-receipts
```

Reviewer 2: this is how I-Ds are referenced in the yaml header part

--- abstract

> This document defines a new verifiable data structure type for COSE Signed Merkle Tree Proofs specifically designed for transaction ledgers produced via Trusted Execution Environments (TEEs), such as the Confidential Consortium Framework ({{CCF}}) to provide stronger tamper-evidence guarantees.

Author: @JAG-UK is this is a sufficient clarification? Should we expand and explain that it's software that's running in the TEE that produces the ledger, or do we think that's reasonably obvious?

~~~ cddl
ccf-proof-element = [
  ; Position of the element
  left: bool

  ; Hash of the proof element: byte string of size HASH_SIZE(32)

>   hash: bstr .size 32;

Author: ```suggestion
  hash: bstr .size 32
```

  return h

verify_inclusion_receipt(inclusion_receipt):
  let label = INCLUSION_PROOF_LABEL

>   let proof = inclusion_receipt.unprotected_header[label] or fail

Author: ```suggestion
  assert(label in inclusion_receipt.unprotected_header)
  let proof = inclusion_receipt.unprotected_header[label]
```

Reviewer 3: I find this proposed change acceptable, but I approved ahead of it with my question/suggestion to clean it up, @achamayou.

>   let proof = inclusion_receipt.unprotected_header[label] or fail

Reviewer 3: Minor, non-blocking: I read the surrounding text and I am not sure what the consequence of pseudocode `or fail`. We discussed maybe it is also worthy of an `assert`? I just wanted to know if `or fail` is like an assert and there should be a hard failure or not.

# Privacy Considerations

See the privacy considerations section of:

*  {{-cose-receipts}}

# Security Considerations

See the security considerations section of:

*  {{-cose-receipts}}

## Trusted Execution Environments

CCF networks of nodes rely on executing in Trusted Execution Environments to secure their function, in particular:

1. The evaluation of registration policies
2. The creation and usage of receipt signing keys

A compromise in the Trusted Execution Environment platform used to execute the network may allow an attacker to produce invalid and incompatible ledger branches.

> Clients can mitigate this risk by regularly fetching attestation information about the TEE instances, available in the ledger and from the network itself, and confirming that the nodes composing the network are running up-to-date, trusted platform components.

Reviewer 4: ```suggestion
Clients can mitigate this risk in two ways: by regularly auditing the consistency of the CCF ledger; and by regularly fetching attestation information about the TEE instances, available in the ledger and from the network itself, and confirming that the nodes composing the network are running up-to-date, trusted platform components.
```

## Operators

> 

Author: We have reasonably complete descriptions of the DR procedure from an operations (https://microsoft.github.io/CCF/main/operations/recovery.html) and governance (https://microsoft.github.io/CCF/main/governance/accept_recovery.html) perspective. I do not know to what extent we want to go in depth here and/or add references. I started short, but happy to expand if desired.

The operator of a CCF network has the ability to start successor networks, with a distinct identity, which endorse the receipts produced by a previous instance.
This functionality is important to provide service continuity in the case of a catastrophic failure of a majority of nodes, but allows a potentially malicious operator to start from a prefix of an earlier ledger.

> Clients can mitigate this risk by checking the latest receipt they hold is present in the ledger of a successor ledger, before they begin making use of it.

Reviewer 4: ```suggestion
Clients can mitigate this risk by auditing the successor ledger and its attestation information, as described above. In particular, clients can check that the latest receipt they hold is present in the successor ledger before they begin making use of it.
```

# Privacy Considerations

See the privacy considerations section of:

*  {{-cose-receipts}}

# Security Considerations

> See the security considerations section of:

Reviewer 2: ```suggestion
The security considerations of {{-cose-receipts}} apply.
```

> *  {{-cose-receipts}}

Reviewer 2: ```suggestion
```
